### PR TITLE
meta-toradex-nxp: add kirkstone support

### PR DESCRIPTION
**Pull request by zach-welch-aquabyte**
Toradex published a `kirkstone-6.x.y` branch, so these patches restore the features that were removed for Mender's `kirkstone` release. Collectively, these changes allow me to run an mender-enabled kirkstone image on a Colibri iMX6ULL SoM with Iris v2.0 carrier board. Upgrades and rollback work as expected.

I suspect there will be issues found during review, but I wanted to publish my first pass and see where things stand.

### CODE REVIEW

**Comment by TheYoctoJester** [comment]
Hi @zach-welch-aquabyte, thanks for the PR. As you state "there will be issues found during review", how deep do you want to go? I can give it a build for sure, but not test on actual hardware. Maybe also @drewmoseley to comment?

File: meta-mender-toradex-nxp/README.md [hunk 1]
```diff
@@ -5,8 +5,6 @@ Mender integration layer for Toradex family of boards.
 The supported and tested boards are:
 - [Toradex Verdin iMX8M Mini](https://hub.mender.io/t/toradex-verdin-imx8m-mini/2908)
 - [Toradex Verdin iMX8M Plus](https://hub.mender.io/t/toradex-verdin-imx8m-plus/5026)
-- [Toradex Colibri i.MX6ULL](https://hub.mender.io/t/toradex-colibri-i-mx6ull/4102/2)
```
Comment (TheYoctoJester) [inline meta-mender-toradex-nxp/README.md:8]: Is this remove intentional?

File: meta-mender-toradex-nxp/classes/image_type_mender_tezi.bbclass [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+    return [
+        OrderedDict({
+          "name": storage_device,
+          "table_type": "gpt",
+              "content": {
+                  "rawfiles": [
+                      {
+                          "dd_options": "bs=8M",
+<<<<<<< HEAD
```
Comment (TheYoctoJester) [inline meta-mender-toradex-nxp/classes/image_type_mender_tezi.bbclass]: This looks like a broken merge, can you please check?

File: meta-mender-toradex-nxp/classes/image_type_mender_tezi.bbclass [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+              "size": 1
+            }
+          }
+        }),
+        OrderedDict({
+          "name": "ubi",
+          "content": {
+            "rawfile": {
+              "dd_options": "bs=8M",
+<<<<<<< HEAD
```
Comment (TheYoctoJester) [inline meta-mender-toradex-nxp/classes/image_type_mender_tezi.bbclass]: Another broken merge?

File: meta-mender-toradex-nxp/classes/image_type_mender_tezi.bbclass [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+    done
+
+    # The first transform strips all folders from the files
+    # The second adds back a subfolder
+    ${IMAGE_CMD_TAR} --transform='s/.*\///' \
+		     --transform 's,^,${IMAGE_LINK_NAME}/,' \
+		     -chf ${IMGDEPLOYDIR}/${IMAGE_NAME}.mender_tezi.tar \
+		     ${WORKDIR}/image-json/image.json ${DEPLOY_DIR_IMAGE}/mender-tezi-metadata/* \
+		     $uboot_files \
+<<<<<<< HEAD
```
Comment (TheYoctoJester) [inline meta-mender-toradex-nxp/classes/image_type_mender_tezi.bbclass]: Another broken merge?

**Review by TheYoctoJester — COMMENTED** [review]
First review:
- I think there are a number of broken merges
- it might be useful to split out the various board integration configurations into separate commits for more clarity when inspecting history, but thats more of a suggestion.
- @drewmoseley can you please review?

**Comment by drewmoseley** [comment]
I suspect we want to wait until the 6.0 release is official. I know there are still some issues with some boards.
Do you have a urgent need for this now?

File: meta-mender-toradex-nxp/README.md (hunk 1 above)
Comment (zach-welch-aquabyte) [reply meta-mender-toradex-nxp/README.md:8]: This commit was a plain `git revert` of a previous commit  (43d7a6bf235e69b288700c297e15daba9ec7c305), as referenced in the commit message. @drewmoseley was the one that made the original commit.

File: meta-mender-toradex-nxp/classes/image_type_mender_tezi.bbclass (hunk 2 above)
Comment (zach-welch-aquabyte) [reply meta-mender-toradex-nxp/classes/image_type_mender_tezi.bbclass]: Again, this was present in the original commit that I reverted. These artifacts were introduced in earlier commits that I am fairly certain never got tested.

I fixed these issues in a subsequent atomic commit (440a192d7dd104f452fc3023b286827d4f156418).

File: meta-mender-toradex-nxp/classes/image_type_mender_tezi.bbclass (hunk 3 above)
Comment (zach-welch-aquabyte) [reply meta-mender-toradex-nxp/classes/image_type_mender_tezi.bbclass]: Same as above.  Reverted, then fixed in the later commit.

File: meta-mender-toradex-nxp/classes/image_type_mender_tezi.bbclass (hunk 4 above)
Comment (zach-welch-aquabyte) [reply meta-mender-toradex-nxp/classes/image_type_mender_tezi.bbclass]: Same as above.  Reverted, then fixed in the later commit.

**Comment by zach-welch-aquabyte** [comment]
> First review:
> 
> * I think there are a number of broken merges

As mentioned above, these have been fixed on this branch.  My builds successfully  produce `mender_tezi` images.

> * it might be useful to split out the various board integration configurations into separate commits for more clarity when inspecting history, but thats more of a suggestion.

The big commit was a revert. I agree that it would have been nicer to remove (and re-add) functionality in smaller chunks, but I thought it was best practice to just do a wholesale revert of the original removals.

> I suspect we want to wait until the 6.0 release is official. I know there are still some issues with some boards. Do you have a urgent need for this now?

The `meta-toradex-nxp` layer only contains `dunfell-5.x.y` branch, so I assume that is the official release branch. So either there is not the official release branch for `dunfell` or the `kirkstone-6.x.y` branch is also an official release branch?  Or are you looking for a tag?  Their latest branch has a new commit that contains the words "Matrix Build Status: SUCCESS", but I cannot reach the jenkins server referenced in that commit to see what exactly that means.

We still have a fair amount of work to do on this, so merging this is not urgent at present.  However, the `meta-mender-community` layer presently only contains a `kirkstone-next` branch. This branch makes it possible to move forward with resolving any further issues that might arise before such a time as it is promoted to the final `kirkstone` branch.

**Comment by zach-welch-aquabyte** [comment]
Another incremental improvement that might be made would be to add a dynamic layer for the `meta-toradex-*` support, such that the base community layer may continue to be used with the meta-freescale layers.  That said, such a change would be a very disruptive  that I think needs to be made through many smaller patches... in a new PR.  Meanwhile, this branch brings the layer back to the state it existed in `dunfell` and allows it to be tested or used more broadly.

**Comment by TheYoctoJester** [comment]
@zach-welch-aquabyte thanks for the elaboration. While I can certainly understand the desire to wait for the official `kirkstone` release by Toradex, my personal opinion matches yours - anything that we can have and works at least to some extent is better than having a better thing in the future and nothing now. Hence I will merge, and we shall sort out the forwarding to `kirkstone` once it arrives.

**Pull request merged**